prunning_implementation/prunning.py

```diff
@@ -0,0 +1,99 @@
+import babeltrace
+from babeltrace import CTFWriter
+import sys
+import tempfile
+from collections import Counter
+from collections import deque
+
+
+#############################################################################
+#    Function to Prunn the data and put in a deque                          #
+#    @param trace_path: The path to the stream file                #
+#           capacity: The capacity of the deque                             #
+############################################################################
+
+def getPrunnedData(trace_path, capacity):
+    
+    myDeque = deque(maxlen = capacity)
+    trace_collection = babeltrace.TraceCollection()
+    
+    # Raise an exception
+    if trace_collection.add_trace(trace_path,'ctf') is None:
+        raise RuntimeError('Cannot add trace')
+    
+    # Gather all the information and discard what you dont easy.
+    # print(trace_collection)
```

Comment:
Delete commented-out code, or control when it gets used. For example, see `experiment.py` use of the `verbose` flag.